Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/ShiftController.cs

```diff
@@ -542,6 +547,32 @@ internal async Task ProcessShiftsAsync(string isRequestFromLogicApp)
             this.telemetryClient.TrackTrace($"ShiftController - ProcessShiftEntitiesBatchAsync ended at: {DateTime.UtcNow.ToString("O", CultureInfo.InvariantCulture)}");
         }
 
+        /// <summary>
+        /// Remove any shifts that do not start before the provided end date.
+        /// </summary>
+        /// <param name="shifts">The shifts to filter.</param>
+        /// <param name="queryStartDate">The date the shifts must start after.</param>
+        /// <param name="queryEndDate">The date the shifts must start before.</param>
+        /// <returns>A list of filtered shifts.</returns>
+        private List<UpcomingShiftsResponse.ScheduleShift> RemoveShiftsWithWrongStartDate(List<UpcomingShiftsResponse.ScheduleShift> shifts, string queryStartDate, string queryEndDate)

```

Comment:
This function appears to be identical to the one in the OpenShiftController, consider moving it to a helper class of some kind and sharing it.